Source/WebCore/rendering/mathml/RenderMathMLRoot.cpp

```diff
@@ -166,6 +166,9 @@ void RenderMathMLRoot::computePreferredLogicalWidths()
         return;
     }
 
+    // Reset the radical operator to ensure it's sized for current content.
+    m_radicalOperator.reset(style());
+
```

Comment:
Can you explain why the preferred width depends on the list of children? IIUC, MathOperator will always use the max width of available glyphs / assembly.

`RenderMathMLRoot::styleDidChange()` already handles the style change. For child list changes, we probably want to override `MathMLRowElement::childrenChanged` in `MathMLRootElement` so that it tells the render class to reset `m_radicalOperator`. And maybe set `RenderObject::setNeedsLayout()` (or `RenderObject::setNeedsLayoutAndPreferredWidthsUpdate()`) ?